app/lib/report_helper.rb

```diff
@@ -613,20 +595,23 @@ def validate_names!(taxon_counts_2d)
     end
     missing_names = Set.new
     missing_parents = {}
-    taxon_counts_2d.each do |tax_id, tax_info|
+
+    tax_2d.each do |tax_id, tax_info|
       level_str = level_name(tax_info['tax_level'])
       if tax_id < 0
         # Usually -1 means accession number did not resolve to species.
         # TODO: Can we keep the accession numbers to show in these cases?
         tax_info['name'] = "All taxa with neither family nor genus classification"
-        if tax_id < TaxonLineage::INVALID_CALL_BASE_ID && (tax_info['tax_level'] == TaxonCount::TAX_LEVEL_GENUS || tax_info['tax_level'] == TaxonCount::TAX_LEVEL_SPECIES)
-          parent_taxid = -(tax_id % TaxonLineage::INVALID_CALL_BASE_ID)
-          if taxon_counts_2d[parent_taxid]
-            parent_name = taxon_counts_2d[parent_taxid]['name']
-            parent_level = level_name(taxon_counts_2d[parent_taxid]['tax_level'])
+
+        if tax_id < TaxonLineage::INVALID_CALL_BASE_ID && species_or_genus(tax_info['tax_level'])
+          parent_id = convert_neg_taxid(tax_id)
+          if tax_2d[parent_id]
+            tax_info["superkingdom_taxid"] = tax_2d[parent_id]["superkingdom_taxid"]
```

Comment:
This line should no longer be necessary, now that pr.update_superkingdom was fixed.